### ISSUE TITLE
NET-1613 Fix S2701 FN: add support for xUnit Assert.True() and Assert.False()

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#8824]
### Description

Related to #8823.
When promoting the rule to SonarWay we decided, in the context of xUnit, to not flag `Assert.True` and `Assert.False` to avoid potential FPs.
This is because `Assert.Fail` was only introduced in 2020 with xUnit version `2.4.2` and before this, developers used `Assert.True(false, message)` and `Assert.False(true, message)` as workarounds to simulate the functionality of `Assert.Fail()`.

The rule is not smart enough to check the boolean value passed as parameter and compare it with the invocation thus we introduced two FN:
```cs
Assert.True(true, someMessage); // FN
Assert.False(false, someMessage); // FN
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1613](https://sonarsource.atlassian.net/browse/NET-1613)

[NET-1613]: https://sonarsource.atlassian.net/browse/NET-1613?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
